### PR TITLE
Refactor useShopSession hook

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Replace `useCart` with `useShopSession` hook.

Create a `ShopSessionContext` to avoid duplicate requests.

Avoid running ShopSession queries in SSR mode.

Implement "get or create shop session" client side.

Fix bug in shop session query mock.

Remove unused shop session helper.

Add option to get and save shop session ID from service.

Fix bug in shop session get method, return null if not found.

## Justify why they are needed

I believe we should aim to use Apollo Client for all client side fetching => or use something else :)

This approach will let us keep an active query on the page that we can invalidate e.g. when an item is added to cart so it updates automatically.

Had to add `eslint-import-resolver-typescript` since husky and VSCode were reporting errors related to `order/import`.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

[GRW-1335]: https://hedvig.atlassian.net/browse/GRW-1335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #379** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/379" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/379?utm_source=stack-comment).

File: apps/store/src/services/shopSession/useShopSession.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    shopSessionId,
+    onCompleted: ({ shopSession }) => {
+      if (shopSession.countryCode !== countryCode) createShopSession()
+    },
+  })
+
+  return useMemo(() => {
+    const shopSessionCountryCode = result.data?.shopSession?.countryCode
+    return {
+      ...result,
```
Comment (alebedev) [inline apps/store/src/services/shopSession/useShopSession.ts:38]: I see we may be returning result from query while waiting for mutation.  I'll see what can be improved

File: apps/store/src/services/shopSession/ShopSessionService.ts [hunk 2]
```diff
@@ -15,10 +15,20 @@ export class ShopSessionService {
     private readonly apolloClient: ApolloClient<unknown>,
   ) {}
 
+  public get shopSessionId() {
```
Comment (alebedev) [inline apps/store/src/services/shopSession/ShopSessionService.ts:18]: Nit: `getShopSessionId()` ?

This is currently the only getter in racoon.  I feed that getters/setters make code less obvious without much improvement in developer experience

File: apps/store/src/services/shopSession/ShopSessionService.ts [hunk 3]
```diff
@@ -15,10 +15,20 @@ export class ShopSessionService {
...
+    return this.persister.fetch()
+  }
+
+  public save(shopSessionId: string) {
```
Comment (alebedev) [inline apps/store/src/services/shopSession/ShopSessionService.ts:22]: Nit: `saveShopSessionId(..)`?  We're not saving whole session

File: apps/store/src/services/shopSession/useShopSession.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/services/shopSession/useShopSession.ts:38]: Checked Apollo API, this part is fine

Comment (robinandeer) [reply apps/store/src/services/shopSession/useShopSession.ts:38]: 👍

File: apps/store/src/services/shopSession/ShopSessionService.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/services/shopSession/ShopSessionService.ts:18]: ahh right, yeah we can change that :)

**Review by KarnellSchultz — APPROVED** [review]
LGTM 🍒

**Pull request merged**